File: app/src/main/java/com/micewine/emu/adapters/AdapterBottomNavigation.java
package com.micewine.emu.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.micewine.emu.fragments.AboutFragment;
import com.micewine.emu.fragments.FileManagerFragment;
import com.micewine.emu.fragments.SettingsFragment;
import com.micewine.emu.fragments.ShortcutsFragment;

public class AdapterBottomNavigation extends FragmentStateAdapter {
    public AdapterBottomNavigation(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return switch (position) {
            case 0 -> new ShortcutsFragment();
            case 1 -> new SettingsFragment();
            case 2 -> new FileManagerFragment();
            case 3 -> new AboutFragment();
            default -> throw new IllegalArgumentException("Invalid Fragment for Position " + position);
        };
    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: app/src/main/java/com/micewine/emu/adapters/AdapterBottomNavigation.kt
package com.micewine.emu.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.micewine.emu.fragments.AboutFragment
import com.micewine.emu.fragments.FileManagerFragment
import com.micewine.emu.fragments.SettingsFragment
import com.micewine.emu.fragments.ShortcutsFragment

class AdapterBottomNavigation(activity: FragmentActivity?) : FragmentStateAdapter(activity!!) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> ShortcutsFragment()
            1 -> SettingsFragment()
            2 -> FileManagerFragment()
            3 -> AboutFragment()
            else -> throw Exception("Invalid Fragment for Position $position")
        }
    }

    override fun getItemCount(): Int {
        return 4
    }
}